webcompat/api/uploads.py

```diff
@@ -26,11 +23,71 @@
 uploads = Blueprint('uploads', __name__, url_prefix='/upload',
                     template_folder='../templates')
 JSON_MIME = 'application/json'
-images = UploadSet('uploads', IMAGES)
-configure_uploads(app, images)
+ALLOWED_FORMATS = ('jpg', 'jpeg', 'jpe', 'png', 'gif', 'bmp')
 
-# limit image uploads to 4MB
-patch_request_class(app, 4 * 1024 * 1024)
+
+class UploadNotAllowed(Exception):
+    """
+    This exception is raised when we try to save a file that is not
+    allowed. Allowed files are defined in ALLOWED_FORMATS
+    """
+
+
+class Upload(object):
+    '''Class that abstracts over saving image and screenshot uploads.
+
+    It performs a simple extension-based validation before saving to the
+    file system. If a file is not allowed, a UploadNotAllowed exception
+    is rased.
+    '''
+    def __init__(self, imagedata):
+        self.imagedata = imagedata
+        self.is_base_64 = (isinstance(imagedata, unicode) and
+                           imagedata.startswith('data:image/'))
+        self.is_filestorage = isinstance(imagedata, FileStorage)
+        self.file_ext = self.get_file_ext(imagedata)
+
+    def get_file_ext(self, imagedata):
+        '''Method to return the file extension.'''
+        if self.is_filestorage:
+            return imagedata.filename.split('.')[-1].lower()
+        if self.is_base_64:
+            # We only expect base64 png screenshot uploads,
```

Comment:
ah interesting.

```
127.0.0.1 - - [22/Dec/2015 16:40:15] "GET /?open=1&url=https%3A%2F%2Fwww.mozilla.org%2Ffr%2F HTTP/1.1" 200 -
127.0.0.1 - - [22/Dec/2015 16:40:17] "GET /api/issues/labels?per_page=100 HTTP/1.1" 200 -
127.0.0.1 - - [22/Dec/2015 16:40:18] "GET /api/issues/category/new HTTP/1.1" 200 -
self.is_base_64    False
self.is_filestorage True
self.file_ext           jpg
127.0.0.1 - - [22/Dec/2015 16:40:39] "POST /issues/new HTTP/1.1" 302 -
127.0.0.1 - - [22/Dec/2015 16:40:39] "GET /thanks/316 HTTP/1.1" 200 -
```

The raise `UploadNotAllowed` is never raised when `self.is_base_64  =  False` in `def get_file_ext(self, imagedata):`.